update-doi-request/src/main/java/no/unit/nva/doi/requests/handlers/UpdateDoiRequestHandler.java

```diff
@@ -0,0 +1,87 @@
+package no.unit.nva.doi.requests.handlers;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import java.util.UUID;
+import no.unit.nva.doi.requests.contants.ServiceConstants;
+import no.unit.nva.doi.requests.exception.BadRequestException;
+import no.unit.nva.doi.requests.handlers.model.ApiTask;
+import no.unit.nva.doi.requests.handlers.model.ApiUpdateDoiResponse;
+import no.unit.nva.doi.requests.model.ApiUpdateDoiRequest;
+import no.unit.nva.doi.requests.service.DoiRequestsService;
+import no.unit.nva.doi.requests.userdetails.UserDetails;
+import nva.commons.exceptions.ApiGatewayException;
+import nva.commons.exceptions.ForbiddenException;
+import nva.commons.handlers.ApiGatewayHandler;
+import nva.commons.handlers.RequestInfo;
+import nva.commons.utils.Environment;
+import org.apache.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class UpdateDoiRequestHandler extends ApiGatewayHandler<ApiUpdateDoiRequest, ApiUpdateDoiResponse> {
+    private static final String LOCATION_TEMPLATE_PUBLICATION = "%s://%s/publication/%s";
+
+    private final DoiRequestsService doiRequestService;
+    private final String apiScheme;
+    private final String apiHost;
+
+    public UpdateDoiRequestHandler(Environment environment, DoiRequestsService doiRequestsService) {
+        super(ApiUpdateDoiRequest.class, environment, initializeLogger());
+        this.apiScheme = environment.readEnv(ServiceConstants.API_SCHEME_ENV_VARIABLE);
+        this.apiHost = environment.readEnv(ServiceConstants.API_HOST_ENV_VARIABLE);
+        this.doiRequestService = doiRequestsService;
+    }
+
+    private static Logger initializeLogger() {
+        return LoggerFactory.getLogger(UpdateDoiRequestHandler.class);
+    }
+
+
+    @Override
+    protected ApiUpdateDoiResponse processInput(ApiUpdateDoiRequest input, RequestInfo requestInfo, Context context)
+        throws ApiGatewayException {
+        input.validate();
+        String username = getUserName(requestInfo);
+        UUID publicationId = getPublicationIdentifier(requestInfo);
+
+        var requestedStatusChange = input.getDOIRequestStatus()
+                .orElseThrow(() -> new BadRequestException("You must request changes to do"));
+
+        try {
+            doiRequestService.updateDoiRequest(publicationId, requestedStatusChange, username);
+        } catch (IllegalArgumentException | IllegalStateException e) {
+            throw new BadRequestException(e.getMessage());
+        }
+
+        return ApiUpdateDoiResponse.newBuilder().withTask(ApiTask.newBuilder()
+            .withHref(getContentLocation(publicationId))
```

Comment:
Dislike we link to the Publication, but since the DoiRequest doesn't have identifiers, we need to delegate responsibility to the owner of the DoiRequest, which is Publication.

This means as an API user they have to fetch the publication (remember its etag), then start the asynchronous process for updating the DoiRequestStatus , and then keep polling publication for updates using the etag until it no longer returns 304 Not Modified. 

@elkebab

Reply:
I don't think we ever want to update a DoiRequest outside the context of a Publication, so this is probably not a big issue.
Since the DoiRequest always is a part of a Publication, this is in reality just a update to a field of an Publication (not a separate object), so you could argue it's more like a PATCH request than a PUT

The polling for updates is cumbersome anyway, as you describe

Reply:
A task I believe we have not solved yet is support of e-tags. Ill create a separate task for this in JIRA to be delivered separately, and we just have to do naive polling in the first version.